addons/utm/models/utm_source.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import uuid
+
+from odoo import fields, models, api, tools, _
+
+
+class UtmSource(models.Model):
+    _name = 'utm.source'
+    _description = 'UTM Source'
+
+    name = fields.Char(string='Source Name', required=True, translate=True)
+    identifier = fields.Char(
+        string='Identifier', readonly=True, index=True, copy=False)
```

Comment:
It's useless to add an index on a field when this field has a unique constraint, as the constraint creates an index...

```
[testdb] # create table test_with_unq_idx(name varchar);
CREATE TABLE
Time: 44.897 ms
[testdb] # create index on test_with_unq_idx(name);
CREATE INDEX
Time: 5.189 ms
[testdb] # alter table test_with_unq_idx add constraint test_ctx unique(name);
ALTER TABLE
Time: 22.967 ms
[testdb] # \d test_with_unq_idx
               Table "public.test_with_unq_idx"
┌────────┬───────────────────┬───────────┬──────────┬─────────┐
│ Column │       Type        │ Collation │ Nullable │ Default │
├────────┼───────────────────┼───────────┼──────────┼─────────┤
│ name   │ character varying │           │          │         │
└────────┴───────────────────┴───────────┴──────────┴─────────┘
Indexes:
    "test_ctx" UNIQUE CONSTRAINT, btree (name)
    "test_with_unq_idx_name_idx" btree (name)
```

Reply:
Didn't know about this, thanks